### DISCUSSION TITLE
Aggregating objects within an array

### DISCUSSION THREAD

**Discussion by toki-developer** [hasura/graphql-engine#10551] [Question]
Is it possible to perform aggregate functions on data structures containing nested objects within arrays?

Example:
We have three tables: projects, applications, and users.
・A project can have multiple applications.
・An application belongs to one user.

I want to retrieve data in the following format.

```
project {
   applications_aggregate {
     aggregate {
       sum {
        user {
          hogehoge
        }
      }
    }
  }
}
```

I want to sum up the "hogehoge" of users by project.

**Comment by codedmart** [comment]
@toki-developer I am assuming we are talking about graphql-engine v2. The query as you written it is not possible as is. I would expect that you could do something like:
```graphql
project {
   applications {
     users_aggregate {
       aggregate {
        sum {
          hogehoge
        }
      }
    }
  }
}
```

**Reply by toki-developer** [reply]
@codedmart 

Thank you for your response.
However, since there is a one-to-one correspondence between applications and users, aggregating the "users" field within the applications data will not yield the desired statistics.
I am trying to calculate application-level statistics.
